sw/airborne/subsystems/navigation/nav_survey_rectangle.c

```diff
@@ -82,20 +83,17 @@ void nav_survey_rectangle_init(uint8_t wp1, uint8_t wp2, float grid, survey_orie
     }
   }
   nav_survey_shift = grid;
+  sweep_var = grid;
   survey_uturn = false;
   LINE_START_FUNCTION;
 }
 
 void nav_survey_rectangle(uint8_t wp1, uint8_t wp2)
 {
-  nav_survey_rectangle_dynamic(wp1, wp2, -1);
-}
-
-void nav_survey_rectangle_dynamic(uint8_t wp1, uint8_t wp2, float grid)
-{
-  if (grid > 0) {
-    if (nav_survey_shift > 0) nav_survey_shift = grid;
-    else nav_survey_shift = -grid;
+  /* Update sweep_width if "extern float sweep" has been updated */
+  if (sweep != nav_survey_shift && sweep != -nav_survey_shift)
```

Comment:
Correct me if I am wrong but nav_survery_shift is a float right? Then this condition will be "always" satisfied.

Reply:
For me, the rest of the PR looks okeish :P.

Reply:
Yes nav_survey_shift is a float, and  so is sweep. I do not think it will always be satisfied, unless I am missing something about float comparison.  I could change it to something like...
Solution 1:
    if (sweep != ABS(nav_survey_shift)) { ....
But I don't think this would change the logic!

Another solution might be to do (i.e. using a flag like you recommended)....
Solution 2:
    #ifdef SURVEY_RECTANGLE_DYNAMIC
        nav_survey_shift = (nav_survey_shift > 0 ? sweep_var : -sweep_var);
    #endif

This would just update nav_survey_shift every iteration, when SURVEY_RECTANGLE_DYNAMIC is defined in the flight_plan header section.

Reply:
Solution 2 is readable and very clean. There are enough resources to allow such an update in every iteration :P.